waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -1145,19 +1146,45 @@ proc mountRlnRelayDynamic*(node: WakuNode,
   node.wakuRlnRelay = rlnPeer
   return ok(true)
 
-proc readPersistentRlnCredentials*(path: string) : RlnMembershipCredentials {.raises: [Defect, OSError, IOError, Exception].} =
-  info "Rln credentials exist in file"
+proc writeRlnCredentials*(path: string, credentials: RlnMembershipCredentials, password: string): RlnRelayResult[void] =
+  info "Storing RLN credentials"
+  var jsonString: string
+  jsonString.toUgly(%credentials)
+  let keyfile = createKeyFileJson(toBytes(jsonString), password)
+  if keyfile.isErr():
+    return err("Error while creating keyfile for RLN credentials")
+  if saveKeyFile(path, keyfile.get()).isErr():
+    return err("Error while saving keyfile for RLN credentials")
+  return ok()
+
+# Attempts decryptions of all keyfiles with the provided password. 
+# If one or more credentials are successfully decrypted, the max(min(index,number_decrypted),0)-th is returned.
+proc readRlnCredentials*(path: string, password: string, index: int = 0): RlnRelayResult[Option[RlnMembershipCredentials]] =
+  info "Reading RLN credentials"
   # With regards to printing the keys, it is purely for debugging purposes so that the user becomes explicitly aware of the current keys in use when nwaku is started.
   # Note that this is only until the RLN contract being used is the one deployed on Goerli testnet.
   # These prints need to omitted once RLN contract is deployed on Ethereum mainnet and using valuable funds for staking.
   waku_rln_membership_credentials_import_duration_seconds.nanosecondTime:
-    let entireRlnCredentialsFile = readFile(path)
 
-    let jsonObject = parseJson(entireRlnCredentialsFile)
-    let deserializedRlnCredentials = to(jsonObject, RlnMembershipCredentials)
-    
-  debug "Deserialized Rln credentials", rlnCredentials=deserializedRlnCredentials
-  return deserializedRlnCredentials
+    try:
+      var decodedKeyfiles = loadKeyFiles(path, password)
+  
+      if decodedKeyfiles.isOk():
+        var decodedRlnCredentials = decodedKeyfiles.get()
+        debug "Successfully decrypted keyfiles for the provided password", numberKeyfilesDecrypted=decodedRlnCredentials.len
+        # We should return the index-th decrypted credential, but we ensure to not overflow
+        let credentialIndex = max(min(index, decodedRlnCredentials.len - 1), 0)
+        debug "Picking credential with (adjusted) index", inputIndex=index, adjustedIndex=credentialIndex
+        let jsonObject = parseJson(string.fromBytes(decodedRlnCredentials[credentialIndex].get()))
+        let deserializedRlnCredentials = to(jsonObject, RlnMembershipCredentials)   
+        debug "Deserialized RLN credentials", rlnCredentials=deserializedRlnCredentials
+        return ok(some(deserializedRlnCredentials))
+      else:
+        debug "Unable to decrypt RLN credentials with provided password. ", error=decodedKeyfiles.error
+        return ok(none(RlnMembershipCredentials))
```

Comment:
Shouldn't this be an error? And return error(xxx). wdyt?

On the other hand perhaps instead of `RlnRelayResult[Option[RlnMembershipCredentials]]` you can use `RlnRelayResult[RlnMembershipCredentials]`? As I understand if it returns ok() it should always reeturn RlnMembershipCredentials right?